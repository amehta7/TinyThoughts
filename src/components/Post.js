import React, { memo } from 'react'

import moment from 'moment'
import DeletePostButton from './DeletePostButton'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { deletePost } from '../store/actions/index'

const Post = memo(({ post, user, onDeletePost }) => {
  return (
    <React.Fragment>
      {post &&
        post.map(
          ({ id, title, content, author, categories, timestamp, authorId }) => {
            return (
              <div className='post' key={id}>
                <div>
                  <h1>{title}</h1>
                  {user ? (
                    <DeletePostButton
                      onConfirm={() => {
                        onDeletePost(id)
                      }}
                    />
                  ) : null}
                  <h3>
                    by{' '}
                    <Link
                      to={`/author/${authorId}`}
                      key={authorId}
                      style={{ color: 'rgb(89, 97, 206)' }}
                    >
                      {author.name} | {categories.title}
                    </Link>{' '}
                    | Posted on{' '}
                    <span>
                      {moment(timestamp).format(
                        'dddd, MMMM Do YYYY, h:mm:ss a'
                      )}
                    </span>
                  </h3>
                  <p>{content}</p>
                </div>
              </div>
            )
          }
        )}
    </React.Fragment>
  )
})

const mapStateToProps = (state) => {
  return {
    posts: state.posts.posts,
    categories: state.posts.categories,
    user: state.users.user,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onDeletePost: (postId) => dispatch(deletePost(postId)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Post)
