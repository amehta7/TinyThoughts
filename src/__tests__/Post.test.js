import React from "react";
import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import Post from "../components/Post";

const mockStore = configureStore([thunk]);

describe("Post component", () => {
  it("Renders a post correctly", () => {
    const store = mockStore({
      users: {
        user: { id: 1, name: "Joe Mockery" }
      },
      posts: {
        categories: [
          { id: 1, title: "JavaScript" },
          { id: 2, title: "Python" }
        ]
      },
      errors: {
        error: false
      }
    });

    const props = {
      post: {
        id: 2,
        title: "This is an amazing post",
        author: { id: 1, name: "Joe Mockery" },
        content:
          "This is such an incredible post that I almost forgot to post it",
        categories: { id: 1, title: "Python" },
        timestamp: "2020-03-25T13:39:30.752Z"
      }
    };

    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter keyLength={0}>
          <Post {...props} />
        </MemoryRouter>
      </Provider>
    );

    expect(wrapper.find(Post)).toMatchSnapshot();
  });
});
