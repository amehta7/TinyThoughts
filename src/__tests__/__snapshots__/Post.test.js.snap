// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component Renders a post correctly`] = `
<Memo()
  post={
    Object {
      "author": Object {
        "id": 1,
        "name": "Joe Mockery",
      },
      "categories": Object {
        "id": 1,
        "title": "Python",
      },
      "content": "This is such an incredible post that I almost forgot to post it",
      "id": 2,
      "timestamp": "2020-03-25T13:39:30.752Z",
      "title": "This is an amazing post",
    }
  }
>
  <div
    className="post"
  >
    <h1>
      This is an amazing post
    </h1>
    <h3>
      by 
      <Link
        to="/author/1"
      >
        <LinkAnchor
          href="/author/1"
          navigate={[Function]}
        >
          <a
            href="/author/1"
            onClick={[Function]}
          >
            Joe Mockery
          </a>
        </LinkAnchor>
      </Link>
       |
       
      <Link
        to="/category/1"
      >
        <LinkAnchor
          href="/category/1"
          navigate={[Function]}
        >
          <a
            href="/category/1"
            onClick={[Function]}
          >
            Python
          </a>
        </LinkAnchor>
      </Link>
       | Posted on 
      Wednesday, March 25th 2020, 7:09:30 pm
    </h3>
    <p>
      This is such an incredible post that I almost forgot to post it
    </p>
    <DeletePostButton
      onConfirm={[Function]}
    >
      <div
        className="delete-post-btn"
      >
        <button
          className="delete-btn"
          onClick={[Function]}
        >
          Delete?
        </button>
      </div>
    </DeletePostButton>
  </div>
</Memo()>
`;
